GROMACS-LS-2016.3 fix compiler flags

This patch removes the -march=native flag, so that EB can set the requested architecure.
Also remove -ffast-math to avoid problems with floating point subnormals.

Author: Oliver Stueker <oliver.stueker@ace-net.ca>
Date:   2024-09-17

diff -ru gromacs-ls-2016.3.orig/src/contrib/mdstress-library/CMakeLists.txt gromacs-ls-2016.3/src/contrib/mdstress-library/CMakeLists.txt
--- gromacs-ls-2016.3.orig/src/contrib/mdstress-library/CMakeLists.txt	2019-12-28 11:13:57.000000000 -0330
+++ gromacs-ls-2016.3/src/contrib/mdstress-library/CMakeLists.txt	2024-09-17 10:32:52.127182300 -0230
@@ -11,7 +11,7 @@
         "Choose the type of build, options are: Debug Release RelWithDebInfo MinSizeRel." FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
-set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG -ffast-math -march=native" CACHE STRING
+set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG " CACHE STRING
     "Flags used by the compiler during release builds." FORCE)
 
 FILE(GLOB MDSTRESS_SOURCES_H   include/mdstress/*.h)
